From 252ecd4663874793859b4bd008aa98ba4c466c34 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Tue, 9 Jul 2024 12:58:01 -0700
Subject: [PATCH] Add clear cert path

This is intended to be patched by the distro so says the comment for
the file. Have the backup /var/cache path used in case /etc isn't
available.

Signed-off-by: William Douglas <william.douglas@intel.com>
---
 src/requests/certs.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/requests/certs.py b/src/requests/certs.py
index be422c3..9215308 100644
--- a/src/requests/certs.py
+++ b/src/requests/certs.py
@@ -11,7 +11,12 @@ If you are packaging Requests, e.g., for a Linux distribution or a managed
 environment, you can change the definition of where() to return a separately
 packaged CA bundle.
 """
-from certifi import where
+import os
+def where():
+    etc_cert = '/etc/ssl/certs/ca-certificates.crt'
+    if os.path.exists(etc_cert):
+        return etc_cert
+    return '/var/cache/ca-certs/anchors/ca-certificates.crt'
 
 if __name__ == "__main__":
     print(where())
-- 
2.45.1

